tests/test_backend.py

```diff
@@ -201,8 +250,42 @@ def test_service_types_simple_cached(
             assert mock_be1.call_count == 2
             assert service_types == single_service_type
 
+    def test_service_types_skips_unsupported_backend(
+        self,
+        multi_backend_connection,
+        config,
+        catalog,
+        backend1,
+        backend2,
+        json_capabilities_no_endpoints,
+        json_capabilities_with_service_types_supported,
+        requests_mock,
+    ):
+        """Given 2 backends and only 1 backend support secondary services, as states in its capabilities,
+            when the aggregator fulfills a request to list the service types,
+            then the aggregator skips the unsupported backend.
+        """
+        single_service_type = self.SERVICE_TYPES_ONLT_WMTS
+        mock_b1_service_types = requests_mock.get(backend1 + "/service_types", status_code=500) # Not supported
+        mock_b2_service_types = requests_mock.get(backend2 + "/service_types", json=single_service_type)
+        # We are testing that the aggregator checks if the backend supports GET /service_types, so we have to mock that up too.
+        # We want to verify that the capabilities are actually queried.
+        mock_b1_capabilities = requests_mock.get(backend1 + "/", json=json_capabilities_no_endpoints)
```

Comment:
FYI: conceptually it makes more sense to first mock the capabilities request `/` and then `/services` (because that's the order clients use).

Reply:
Went through all tests for secondary services and adapted the order to keep things consistent.